src/metaswitch/clearwater/cluster_manager/etcd_synchronizer.py

```diff
@@ -77,7 +77,8 @@ def main(self):
                 # necessary, set this node to WAITING_TO_LEAVE. Otherwise, kick
                 # the FSM.
                 if (self._leaving_flag and
-                        cluster_info.can_leave(self.force_leave)):
+                    cluster_info.local_state(self._ip) != constants.WAITING_TO_LEAVE and
+                    cluster_info.can_leave(self.force_leave)):
```

Comment:
We discussed in some details (thank you being patient with me!). I'm happy that this is the right pragmatic fix for the time being. However can you add some comments explaining how the `_leaving_flag` can get out of sync with etcd. 

I also think it would be useful to rename this to `_leave_requested` or something similar.